types/types.go

```diff
@@ -107,12 +107,13 @@ type ImageDestination interface {
 	Reference() ImageReference
 	// FIXME? This should also receive a MIME type if known, to differentiate between schema versions.
 	PutManifest([]byte) error
-	// PutBlob writes contents of stream as a blob identified by digest.
+	// PutBlob writes contents of stream and returns its computed digest and size (both if can be computed).
```

Comment:
… actually, that `ReaderForDescriptor` in public interface and not returning the data won’t work for the “Docker registry already contains the blob, we have used `HEAD` to get the metadata” case. It would still be an useful utility for the `PutBlob` implementations, but you are right that this needs to be returned by the method.